oak_protobuf_idl/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#![feature(associated_type_defaults)]
+#![feature(never_type)]
+#![feature(try_blocks)]
+
+mod proto {
+    #![allow(dead_code)]
+    include!(concat!(env!("OUT_DIR"), "/oak.protobuf_idl.rs"));
+}
+
+pub use oak_idl::{Status, StatusCode};
+pub use proto::{response, Request, Response};
+
+use prost::Message;
+
+/// A message-oriented transport that allows performing invocations.
+pub trait Transport {
+    /// Type representing any transport-specific errors. By default, the transport is assumed to be
+    /// infallible.
+    type Error = !;
+    fn invoke(&mut self, request_bytes: &[u8]) -> Result<Vec<u8>, Self::Error>;
+}
+
+#[async_trait::async_trait]
+pub trait AsyncTransport {
```

Comment:
Will you add a 1-line doc comment here too?

Reply:
Sure, added on the methods too.